crossbeam-channel/src/flavors/zero.rs

```diff
@@ -177,7 +190,7 @@ impl<T> Channel<T> {
             // heap-allocated packet.
             packet.wait_ready();
             let msg = packet.msg.get().replace(None).unwrap();
-            drop(Box::from_raw(packet as *const Packet<T> as *mut Packet<T>));
+            drop(Box::from_raw(token.zero.0 as *mut Packet<T>));
             Ok(msg)
```

Comment:
Another stacked borrows violation that was fixed by this PR is (While `packet` (an immutable reference to heap-allocated Packet) was still valid, Box::from_raw was called)):

```
error: Undefined Behavior: trying to reborrow for Unique at alloc5094412+0x8, but parent tag <14690213> does not have an appropriate item in the borrow stack
    --> /Users/taiki/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/src/rust/library/alloc/src/boxed.rs:977:9
     |
977  |         Box(unsafe { Unique::new_unchecked(raw) }, alloc)
     |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ trying to reborrow for Unique at alloc5094412+0x8, but parent tag <14690213> does not have an appropriate item in the borrow stack
     |
     = help: this indicates a potential bug in the program: it performed an invalid operation, but the rules it violated are still experimental
     = help: see https://github.com/rust-lang/unsafe-code-guidelines/blob/master/wip/stacked-borrows.md for further information
             
     = note: inside `std::boxed::Box::<crossbeam_channel::flavors::zero::Packet<i32>>::from_raw_in` at /Users/taiki/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/src/rust/library/alloc/src/boxed.rs:977:9
     = note: inside `std::boxed::Box::<crossbeam_channel::flavors::zero::Packet<i32>>::from_raw` at /Users/taiki/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/src/rust/library/alloc/src/boxed.rs:921:18
     = note: inside `crossbeam_channel::flavors::zero::Channel::<i32>::read` at /Users/taiki/projects/crossbeam/crossbeam-channel/src/flavors/zero.rs:193:18
     = note: inside `crossbeam_channel::flavors::zero::Channel::<i32>::recv` at /Users/taiki/projects/crossbeam/crossbeam-channel/src/flavors/zero.rs:302:24
     = note: inside `crossbeam_channel::Receiver::<i32>::recv` at /Users/taiki/projects/crossbeam/crossbeam-channel/src/channel.rs:802:43
note: inside `Chan::<i32>::recv` at crossbeam-channel/tests/golang.rs:64:9
    --> crossbeam-channel/tests/golang.rs:64:9
     |
64   |         r.recv().ok()
     |         ^^^^^^^^
note: inside closure at crossbeam-channel/tests/golang.rs:1420:28
    --> crossbeam-channel/tests/golang.rs:1420:28
     |
1420 |                     l[i] = c.recv().unwrap();
     |                            ^
```